src/components/ha-camera-stream.ts

```diff
@@ -0,0 +1,153 @@
+import { property, UpdatingElement, PropertyValues } from "lit-element";
+
+import computeStateName from "../common/entity/compute_state_name";
+import { HomeAssistant, CameraEntity } from "../types";
+import { fireEvent } from "../common/dom/fire_event";
+import { fetchStreamUrl, computeMJPEGStreamUrl } from "../data/camera";
+
+type HLSModule = typeof import("hls.js");
+
+class HaCameraStream extends UpdatingElement {
+  @property() public hass?: HomeAssistant;
+  @property() public stateObj?: CameraEntity;
+  @property() public showControls: boolean = false;
+  private _hlsPolyfillInstance?: Hls;
+
+  public connectedCallback() {
+    super.connectedCallback();
+    // @ts-ignore
+    if (!this._hasRequestedUpdate) {
+      this.requestUpdate();
+    }
+  }
+
+  public disconnectedCallback() {
+    super.disconnectedCallback();
+    this._teardownPlayback();
+  }
+
+  protected update(changedProps: PropertyValues) {
+    super.update(changedProps);
+    if (this.lastChild) {
+      return;
+    }
+
+    const oldState = changedProps.get("stateObj") as this["stateObj"];
+    const oldEntityId = oldState ? oldState.entity_id : undefined;
+    const curEntityId = this.stateObj ? this.stateObj.entity_id : undefined;
+
+    // Same entity, ignore.
+    if (curEntityId === oldEntityId) {
+      return;
+    }
+
+    // Tear down if we have something and we need to build it up
+    if (oldEntityId) {
+      this._teardownPlayback();
+    }
+
+    if (curEntityId) {
+      this._startPlayback();
+    }
+  }
+
+  private async _startPlayback(): Promise<void> {
+    if (!this.stateObj) {
+      return;
+    }
+
+    if (!this.hass!.config.components.includes("stream")) {
```

Comment:
I miss one change that we currently have, [`!supportsFeature(this.stateObj, CAMERA_SUPPORT_STREAM)`](https://github.com/home-assistant/home-assistant-polymer/blob/dev/src/dialogs/more-info/controls/more-info-camera.ts#L88)